packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -85,30 +95,33 @@ export const useMenu = (props: MenuProps, ref: React.Ref<HTMLElement>, defaultPr
   );
 
   useMenuPopup(state);
-  useOnClickOutside({ element: document, refs: [state.menuPopupRef, triggerRef], callback: () => setOpen(false) });
+  useOnClickOutside({
+    element: document,
+    refs: [state.menuPopupRef, triggerRef],
+    callback: () => setOpen(false),
+  });
+
   return state;
 };
 
 const useMenuPopup = (state: MenuState) => {
-  // TODO use Popper for the popup ref
-  const menuPopupRef = React.useRef<HTMLElement>(null) as React.MutableRefObject<HTMLElement>;
-  state.menuPopupRef = menuPopupRef;
   state.menuPopup.children = (Component, originalProps) => {
     const newProps = { 'aria-labelledby': state.triggerId, ...originalProps };
 
-    if (state.onHover && !state.onContext) {
-      newProps.onMouseEnter = (e: React.MouseEvent<HTMLElement>) => {
+    newProps.onMouseEnter = (e: React.MouseEvent<HTMLElement>) => {
+      if (state.onHover && !state.onContext) {
         state.setOpen(true);
-        originalProps?.onMouseEnter?.(e);
-      };
-      newProps.onMouseLeave = (e: React.MouseEvent<HTMLElement>) => {
-        state.setOpen(false);
-        originalProps?.onMouseLeave?.(e);
-      };
-    }
+      }
+
+      originalProps?.onMouseEnter?.(e);
+    };
```

Comment:
Shouldn't use `useCallback` in this case?

Reply:
this one not really, since it's not actually a react component but a render function